shreqt/core.py

```diff
@@ -20,6 +20,15 @@ def __init__(self, db_type: DBType):
 
         # self.connection.ping() # Should not be in constructor
 
+    def freeze(self, test_fun):
+        def wrapper(*args, **kwargs):
+            with self.connection.connect(autocommit=False) as conn:
+                res = test_fun(conn, *args, **kwargs)
```

Comment:
What about:
```
yield test_fun(conn, *args, **kwargs)
conn.rollback()
```

Reply:
That would turn the function into generator and in order to call `rollback` we'd have to call the function twice in order to return to statement after yield. E.g.: this test:

```
result = some_test()
assert result == 1
assert cursor_mock.rollback.call_count == 1
```

Turns into:

```
result = some_test()
assert next(result) == 1
try:
    next(result)
except StopIteration:
    pass
assert cursor_mock.rollback.call_count == 1
```

or into:

```
result = list(some_test())[0]
assert result == 1
assert cursor_mock.rollback.call_count == 1
```

Reply:
That's right, I forgot that that it's used as a decorator, not as a `pytest` fixture